src/icp_canister_backend/tests/test.rs

```diff
@@ -522,3 +522,58 @@ fn test_user_deposit_tracking() {
         "Remaining deposit should have ID 1"
     );
 }
+
+#[test]
+fn test_get_deposit() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user = Principal::from_text("xkbqi-2qaaa-aaaah-qbpqq-cai").unwrap();
+    let timelock: u64 = 86400;
+    let deposit_amount = Nat::from(100_000_000u64);
+
+    // Try to get non-existent deposit
+    let get_deposit_result = pic
+        .query_call(
+            canister_id,
+            user,
+            "get_deposit",
+            encode_args((999u64,)).unwrap(),
+        )
+        .expect("Failed to call get_deposit");
+    let non_existent_deposit: Option<Deposit> = decode_one(&get_deposit_result).unwrap();
+    assert!(
+        non_existent_deposit.is_none(),
+        "Non-existent deposit should return None"
+    );
+
+    // Create a deposit
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        user,
+        deposit_amount.clone(),
+        timelock,
+    );
+
+    // Get the created deposit
+    let get_deposit_result = pic
+        .query_call(
+            canister_id,
+            user,
+            "get_deposit",
+            encode_args((0u64,)).unwrap(),
+        )
+        .expect("Failed to call get_deposit");
+    let deposit: Option<Deposit> = decode_one(&get_deposit_result).unwrap();
+
+    assert!(deposit.is_some(), "Deposit should exist");
+    let deposit = deposit.unwrap();
+    assert_eq!(
+        deposit.amount,
+        deposit_amount.clone() - TRANSFER_FEE.clone(),
+        "Deposit should have correct amount"
+    );
+
+    let expected_unlocktime = deposit.unlocktime - (timelock * 1_000_000_000);
+    assert_eq!(deposit.unlocktime, expected_unlocktime + (timelock * 1_000_000_000), "Deposit unlock time should match exactly");
```

Comment:
This check doesn’t make any sense.
Before creating a deposit get the time and than add timelock to it. This will be expected value.
Then compare it with actual unlocktime. Allow +- 3 sec difference.